deps/wazuh_testing/wazuh_testing/qa_docs/doc_generator.py

```diff
@@ -1,121 +1,150 @@
-"""
-brief: Wazuh DocGenerator tool.
-copyright: Copyright (C) 2015-2021, Wazuh Inc.
-date: August 02, 2021
-license: This program is free software; you can redistribute it
-         and/or modify it under the terms of the GNU General Public
-         License (version 2) as published by the FSF - Free Software Foundation.
-"""
+# Copyright (C) 2015-2021, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
 
 import os
 import re
 import json
 import yaml
-from wazuh_testing.qa_docs.lib.config import Config, mode
+
+from wazuh_testing.qa_docs.lib.config import Mode
 from wazuh_testing.qa_docs.lib.code_parser import CodeParser
 from wazuh_testing.qa_docs.lib.utils import clean_folder
 from wazuh_testing.qa_docs import QADOCS_LOGGER
 from wazuh_testing.tools.logging import Logging
 from wazuh_testing.tools.exceptions import QAValueError
 
+
 class DocGenerator:
-    """
-    brief: Main class of DocGenerator tool.
-    It´s in charge of walk every test file, and every group file to dump the parsed documentation.
+    """Main class of DocGenerator tool.
+
+    It is in charge of walk every test file, and every group file to dump the parsed documentation.
+    Every folder is checked so they are ignored when the path matches. Then, every test from folders not ignored
+    that matches a include regex, is parsed.
+
+    Attributes:
+        conf: A `Config` instance with data loaded from config file.
+        parser: A `CodeParser` instance with parsing utilities.
+        __id_counter: An integer that counts the test/group ID when it is created.
```

Comment:
Perhaps it would be better to indicate the arguments/attributes type in parentheses next to their name:

```suggestion
        __id_counter (int): Counts the test/group ID when it is created.
```

Reply:
Solved in https://github.com/wazuh/wazuh-qa/pull/1907/commits/bc05b1c40cf221a3dd219cd9cb8c8560793f3413